File: sorting/bubble-sort/bubble-sort.py

list = [7, 5, 8, 12, 2, 9, 10]

def bubble_sorting(list):
  
  for i in range(len(list)):

    for j in range(0,len(list)-1-i):
      
      if list[j] > list[j+1]:
       list[j],list[j+1] = list[j+1],list[j]

  return list
  
bubble_sorting(list)

# new list [ 2, 5, 7, 8, 9, 10, 12]

print(bubble_sorting(list))

File: sorting/bubble-sort/js/bubble-sort.js
const arr = [7, 5, 8, 12, 2, 9, 10];

function bubbleSorting(arr) {
  let temp;
  for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arr.length-1-i; j++) {
      if (arr[j] > arr[j+1]) {
        temp = arr[j];
        arr[j] = arr[j+1];
        arr[j+1] = temp;
      }
    }
  }
  return arr;
}

bubbleSorting(arr);

// new arr [ 2, 5, 7, 8, 9, 10, 12 ]

console.log(bubbleSorting(arr));
